docs/development/design/cockroachdb.md

```diff
@@ -0,0 +1,22 @@
+# Support Cockroach DB
+
+> [Issue #598](https://github.com/redhat-et/issues/598)
+
+## Summary
+
+Support a DB technology that can reduce DB downtime of the service and allow the DB to horizontally scale.  
+
+## Proposal
+
+We should support Cockroach DB in addition to Postgres.  Cockroach shards and replicates over several work nodes allowing it to be horizontally scalable and have higher availability than traditional single node databases.  It also support doing zero downtime upgrades of the database.
```

Comment:
Could we support Cockroach in place of Postgres?

Reply:
If we could support both, it would be ideal.  The downside to Cockroach is that it's more resource intensive.  You need more hardware to get the similar performance to Postgres.  So for smaller deployments or on prem deployments, I think Postgres makes alot of sense.